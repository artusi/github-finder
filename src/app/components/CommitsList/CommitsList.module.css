.commitsList {
}
.commitsReport {
}
.idCell {
  max-width: 90px;
  text-align: center;
}
.authorCell {
  max-width: 90px;
}
