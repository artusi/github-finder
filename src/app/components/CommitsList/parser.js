import React from "react";

export const commitReportParser = (commit, styles) => {
  const commitIdCell = {
    customClass: styles.idCell,
    content: (
      <a
        href={commit.url}
        target="_blank"
        rel="noopener noreferrer"
        title={`Go to commit -> ${commit.id}`}
      >
        {commit.id.substring(0, 10)}
      </a>
    )
  };
  const commitDescriptionCell = {
    customClass: styles.descriptionCell,
    content: commit.description
  };
  const commitAuthorCell = {
    customClass: styles.authorCell,
    content: commit.author
  };
  return [commitIdCell, commitDescriptionCell, commitAuthorCell];
};

export default commitReportParser;
