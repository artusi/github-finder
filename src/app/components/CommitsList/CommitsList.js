import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Report } from "ui";
import { commitReportParser } from "./parser";
import styles from "./CommitsList.module.css";

export function CommitsList({ commits, error }) {
  const errorMessage = "Sorry but we couldn't find this user or repository ;-(";

  if (error) return <div className={styles.commitsList}>{errorMessage}</div>;

  const parsed = commits.list.map(item => ({
      id: item.id,
      cells: commitReportParser(item, styles)
    }));

  return (
    <div className={styles.commitsList}>
      <p>List of commits:</p>
      <div className={styles.commitsReport}>
        <Report id="commits-report" list={parsed} />
      </div>
    </div>
  );
}

const mapStateToProps = state => ({
  commits: state.github.currentRepo.commits,
  error: state.github.error
});

CommitsList.propTypes = {
  commits: PropTypes.shape({
    total: PropTypes.number.isRequired,
    list: PropTypes.array.isRequired
  }).isRequired,
  error: PropTypes.bool.isRequired
};

export default connect(mapStateToProps)(CommitsList);
