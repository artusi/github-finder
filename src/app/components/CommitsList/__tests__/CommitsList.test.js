import React from "react";
import { CommitsList } from "../CommitsList";

describe("<CommitsList />", () => {
  const initialProps = {
    error: false,
    commits: {
      total: 2,
      list: [
        {
          id: "e3eff3dc09b004a45aeb19db66d403a482163922",
          description:
            "Merge pull request #3 from adriancmiranda/patch-1\n\nUpdate README.md",
          author: "Rafael Artusi",
          url:
            "https://github.com/artusi/jclip/commit/e3eff3dc09b004a45aeb19db66d403a482163922"
        },
        {
          id: "2bb637c960af94a9c5436849a3aa73dfdf36fc88",
          description: "Update README.md",
          author: "Adrian Miranda",
          url:
            "https://github.com/artusi/jclip/commit/2bb637c960af94a9c5436849a3aa73dfdf36fc88"
        }
      ]
    }
  };

  it("renders", () => {
    const wrapper = shallow(<CommitsList {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
