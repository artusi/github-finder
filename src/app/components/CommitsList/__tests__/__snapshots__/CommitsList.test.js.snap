// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitsList /> renders 1`] = `
<div
  className="commitsList"
>
  <p>
    List of commits:
  </p>
  <div
    className="commitsReport"
  >
    <Report
      customClassName=""
      id="commits-report"
      list={
        Array [
          Object {
            "cells": Array [
              Object {
                "content": <a
                  href="https://github.com/artusi/jclip/commit/e3eff3dc09b004a45aeb19db66d403a482163922"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="Go to commit -> e3eff3dc09b004a45aeb19db66d403a482163922"
                >
                  e3eff3dc09
                </a>,
                "customClass": "idCell",
              },
              Object {
                "content": "Merge pull request #3 from adriancmiranda/patch-1

Update README.md",
                "customClass": "descriptionCell",
              },
              Object {
                "content": "Rafael Artusi",
                "customClass": "authorCell",
              },
            ],
            "id": "e3eff3dc09b004a45aeb19db66d403a482163922",
          },
          Object {
            "cells": Array [
              Object {
                "content": <a
                  href="https://github.com/artusi/jclip/commit/2bb637c960af94a9c5436849a3aa73dfdf36fc88"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="Go to commit -> 2bb637c960af94a9c5436849a3aa73dfdf36fc88"
                >
                  2bb637c960
                </a>,
                "customClass": "idCell",
              },
              Object {
                "content": "Update README.md",
                "customClass": "descriptionCell",
              },
              Object {
                "content": "Adrian Miranda",
                "customClass": "authorCell",
              },
            ],
            "id": "2bb637c960af94a9c5436849a3aa73dfdf36fc88",
          },
        ]
      }
    />
  </div>
</div>
`;
